Sources/SwifterSwift/Foundation/NSAttributedStringExtensions.swift

```diff
@@ -180,4 +180,26 @@ public extension NSAttributedString {
     }
 }
 
+public extension Array where Element: NSAttributedString {
+    /// SwifterSwift: Returns a new `NSAttributedString` by concatenating the elements of the sequence, adding the given separator between each element.
+    ///
+    /// - Parameter separator: An `NSAttributedString` to add between the elements of the sequence.
+    /// - Returns: NSAttributedString with applied attributes.
+    // https://stackoverflow.com/questions/32830519/is-there-joinwithseparator-for-attributed-strings
+    func joined(separator: NSAttributedString) -> NSAttributedString {
+        var isFirst = true
+        return reduce(into: NSMutableAttributedString()) { result, element in
+            if isFirst {
+                isFirst = false
+            } else {
+                result.append(separator)
+            }
+            result.append(element)
+        }
```

Comment:
Here is an implementation suggestion to avoid having to check `isFirst` every iteration
```suggestion
        guard let firstElt = self.first else { return NSMutableAttributedString() }
        var mStr = NSMutableAttributedString(firstElt)
        return dropFirst().reduce(into: mStr) { result, element in
            result.append(separator)
            result.append(element)
        }
```
Maybe not even need a reduce into, a for-loop would be fine :)

Reply:
Hello @LucianoPAlmeida 
Thank you for your suggestion, let's use it.